src/ansi.rs

```diff
@@ -0,0 +1,1912 @@
+    Foreground = 256,
+    /// The background color.
+    Background,
+    /// Color for the cursor itself.
+    Cursor,
+    /// Dim black.
+    DimBlack,
+    /// Dim red.
+    DimRed,
+    /// Dim green.
+    DimGreen,
+    /// Dim yellow.
+    DimYellow,
+    /// Dim blue.
+    DimBlue,
+    /// Dim magenta.
+    DimMagenta,
+    /// Dim cyan.
+    DimCyan,
+    /// Dim white.
+    DimWhite,
+    /// The bright foreground color.
+    BrightForeground,
+    /// Dim foreground.
+    DimForeground,
+}
+
+impl NamedColor {
+    #[must_use]
+    pub fn to_bright(self) -> Self {
+        match self {
+            NamedColor::Foreground => NamedColor::BrightForeground,
+            NamedColor::Black => NamedColor::BrightBlack,
+            NamedColor::Red => NamedColor::BrightRed,
+            NamedColor::Green => NamedColor::BrightGreen,
+            NamedColor::Yellow => NamedColor::BrightYellow,
+            NamedColor::Blue => NamedColor::BrightBlue,
+            NamedColor::Magenta => NamedColor::BrightMagenta,
+            NamedColor::Cyan => NamedColor::BrightCyan,
+            NamedColor::White => NamedColor::BrightWhite,
+            NamedColor::DimForeground => NamedColor::Foreground,
+            NamedColor::DimBlack => NamedColor::Black,
+            NamedColor::DimRed => NamedColor::Red,
+            NamedColor::DimGreen => NamedColor::Green,
+            NamedColor::DimYellow => NamedColor::Yellow,
+            NamedColor::DimBlue => NamedColor::Blue,
+            NamedColor::DimMagenta => NamedColor::Magenta,
+            NamedColor::DimCyan => NamedColor::Cyan,
+            NamedColor::DimWhite => NamedColor::White,
+            val => val,
+        }
+    }
+
+    #[must_use]
+    pub fn to_dim(self) -> Self {
+        match self {
+            NamedColor::Black => NamedColor::DimBlack,
+            NamedColor::Red => NamedColor::DimRed,
+            NamedColor::Green => NamedColor::DimGreen,
+            NamedColor::Yellow => NamedColor::DimYellow,
+            NamedColor::Blue => NamedColor::DimBlue,
+            NamedColor::Magenta => NamedColor::DimMagenta,
+            NamedColor::Cyan => NamedColor::DimCyan,
+            NamedColor::White => NamedColor::DimWhite,
+            NamedColor::Foreground => NamedColor::DimForeground,
+            NamedColor::BrightBlack => NamedColor::Black,
+            NamedColor::BrightRed => NamedColor::Red,
+            NamedColor::BrightGreen => NamedColor::Green,
+            NamedColor::BrightYellow => NamedColor::Yellow,
+            NamedColor::BrightBlue => NamedColor::Blue,
+            NamedColor::BrightMagenta => NamedColor::Magenta,
+            NamedColor::BrightCyan => NamedColor::Cyan,
+            NamedColor::BrightWhite => NamedColor::White,
+            NamedColor::BrightForeground => NamedColor::Foreground,
+            val => val,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Color {
+    Named(NamedColor),
+    Spec(Rgb),
+    Indexed(u8),
+}
```

Comment:
I don't think removing the Serialize/Deserialize flags here is a good idea. I believe this is required for Alacritty's tests? Have you checked or did you just remove it? This module should obviously integrate with Alacritty.

Reply:
I could add the `Serialize` and `Deserialize` derives behind a `ansi-serde` feature flag.

Reply:
The feature flag should be `serde`, not `ansi-serde`.

Reply:
> The feature flag should be `serde`, not `ansi-serde`.

Fixed in https://github.com/alacritty/vte/pull/85/commits/76440a7b375f43bb7c066b940cdc552c692ec6cd.